env0/provider_test.go

```diff
@@ -26,14 +32,83 @@ var testUnitProviders = map[string]func() (*schema.Provider, error){
 
 func runUnitTest(t *testing.T, testCase resource.TestCase, mockFunc func(mockFunc *client.MockApiClientInterface)) {
 	testReporter := utils.TestReporter{T: t}
-
 	ctrl = gomock.NewController(&testReporter)
+	defer ctrl.Finish()
+
+	os.Setenv("ENV0_API_KEY", "value")
+	os.Setenv("ENV0_API_SECRET", "value")
+	defer os.Setenv("ENV0_API_KEY", "")
+	defer os.Setenv("ENV0_API_SECRET", "")
 
 	apiClientMock = client.NewMockApiClientInterface(ctrl)
 	mockFunc(apiClientMock)
 
 	testCase.ProviderFactories = testUnitProviders
 	resource.UnitTest(&testReporter, testCase)
+}
+
+func TestProvider(t *testing.T) {
```

Comment:
wondering the reason why this function is Public and other tests are Private methods.

Reply:
That's how `go` runs tests. 
When you do `go test`, go tests for every function that its name has a `Test` prefix and its signature is `t *testint.T`.

Not sure what's the private methods you refer to 🤔